src/System.IO.Pipelines/StreamExtensions.cs

```diff
@@ -78,9 +64,18 @@ public static IPipeReader AsPipelineReader(this Stream stream, CancellationToken
         /// <param name="stream"></param>
         /// <param name="writer"></param>
         /// <returns></returns>
-        public static Task CopyToAsync(this Stream stream, IPipeWriter writer)
+        public static async Task CopyToAsync(this Stream stream, IPipeWriter writer)
         {
-            return stream.CopyToAsync(new PipelineWriterStream(writer));
+            try
+            {
+                await stream.CopyToAsync(new PipelineWriterStream(writer));
+            }
+            catch (Exception ex)
+            {
+                writer.Complete(ex);
+                return;
+            }
+            writer.Complete();
```

Comment:
I am not sure about that @halter73 in fact maybe it should be removed. If there is no exception is the expected behavior that the pipe is completed. I think that would be more likely to be named CopyToEndAsync or similar.